store/store.go

```diff
@@ -602,6 +612,76 @@ func (s *Store) doRequest(client *http.Client, reqOptions *requestOptions, user
 	return resp, err
 }
 
+
+type decodeFunc func(resp *http.Response) (error, bool)
+
+func (s *Store) retryRequest(client *http.Client, reqOptions *requestOptions, user *auth.UserState, retryStrategy retry.Strategy, decode decodeFunc) error {
```

Comment:
Maybe this could be factored into a function that works when called
inside an explicit retry loop.

Something like:


      // shouldRetry reports whether the request that returned the
      // given HTTP response should be retried.
      func shouldRetry(resp *http.Response) (bool, error)

Then you could use it directly inside explicit loops elsewhere rather than
using the callback style. YMMV!